src/functions/text.js

```diff
@@ -7,17 +7,28 @@ import * as html from "../buildHTML";
 import * as mml from "../buildMathML";
 
 // Non-mathy text, possibly in a font
-const textFunctionFonts = {
-    "\\text": undefined, "\\textrm": "mathrm", "\\textsf": "mathsf",
-    "\\texttt": "mathtt", "\\textnormal": "mathrm", "\\textbf": "mathbf",
+const textFontFamilies = {
+    "\\text": undefined, "\\textrm": "textrm", "\\textsf": "textsf",
+    "\\texttt": "texttt", "\\textnormal": "textrm",
```

Comment:
The reason for using `mathrm`, `mathsf`, etc. here is that that's what's used in `fontMap` in `buildCommon`.  Not quite sure how this was working without updating `fontMap`, it'd be good to figure that out.  Maybe we can get rid of it.  We probably should separate the font family from whether we're in math mode or text mode.  Maybe we should change this to:
```
"\\text": undefined, "\\textrm": "rm", "\\textsf": "sf", ...
```

Same thing for font weights and font shapes as we have both `\\textbf` and `\\mathbf` as well as `\\textit` and `\\mathit` commands.

We'll also need to update `fontMap` if it's still needed.

Reply:
Font map is now strictly for math functions, and old text commands (\rm, \sf, etc.). The reason is the two are treated differently.

Text allows the fontFamily/fontWeight/fontShape to stack. In other words you can't simply use the last function to determine what the font is.

i.e. `\textsf{\textbf{H}}` combines the two to make a bold SansSerif H.

This means that the old fontMap won't work either wont work well for text, or will need to be very verbose (contain every permutation of family, weight, shape).

Math and old font commands change the font as a whole.

i.e. `\mathsf{\mathbf{H}}` will make a bold, non-SansSerif H.

The old fontMap works fine here.


The determination for which route to go with happens here:

https://github.com/Khan/KaTeX/pull/1009/files#diff-91067d2948380956002a57296faf3c18R208

Reply:
Cool.  I didn't realize that the math font commands behaved differently from the text commands.  Thanks for the explanation.